docs-src/docs/guides/walkthrough.md

```diff
@@ -0,0 +1,317 @@
+# An Iterative Walkthrough
+
+This guide aims to take you through three increasingly-complex deployment examples using Layer0. Successive sections build upon the previous ones, and each deployment can be completed either through using the Layer0 CLI directly, or through Terraform using our custom Layer0 Terraform Provider.
+
+We assume that you're using Layer0 v0.9.0 or later. If you have not already installed and configured Layer0, see the [installation guide](/setup/install). If you are running an older version of Layer0, you may need to [upgrade](/setup/upgrade#upgrading-older-versions-of-layer0).
+
+If you intend to deploy services using the Layer0 Terraform Provider, you'll want to make sure that you've [installed](/reference/terraform-plugin/#install) the provider correctly.
+
+**Table of Contents**:
+
+- [Deployment 1](#deployment-1-a-simple-guestbook-app): A Simple Guestbook App
+- [Deployment 2](#deployment-2-guestbook-redis): Guestbook + Redis
+- [Deployment 3](#deployment-3-guestbook-redis-consul): Guestbook + Redis + Consul
+
+
+## Deployment 1: A Simple Guestbook App
+
+In this section you'll learn how different Layer0 commands work together to deploy web applications to the cloud. The example application in this section is a guestbook -- a web application that acts as a simple message board. You can choose to complete this section using either the [Layer0 CLI](#1a-deploy-with-layer0-cli) or [Terraform](#1b-deploy-with-terraform).
```

Comment:
[nit] `to deploy web applications to the cloud....` should be `to deploy applications to the cloud`. We later mention the example is a web application in the same sentence, and Layer0 can be used to deploy other types of apps than just web.